### PR TITLE
contrib: add groupBy and group

### PR DESCRIPTION
**Pull request by vaibhavsagar**

### CODE REVIEW

**Comment by msmorgan** [comment]
Should this be in `contrib`?

**Comment by ahmadsalim** [comment]
@vaibhavsagar Thanks for the PR. Yes, please put them in `contrib`.

**Comment by vaibhavsagar** [comment]
Hi, is this what you had in mind?

File: libs/contrib/Data/List/Group.idr [hunk 1]
```diff
@@ -0,0 +1,24 @@
+module Group
```
Comment (ahmadsalim) [inline libs/contrib/Data/List/Group.idr]: I think this should be `Data.List.Group`

**Comment by ahmadsalim** [comment]
@vaibhavsagar Yeah, thanks. There is a small correction with regards to module name but otherwise it is fine.

**Comment by vaibhavsagar** [comment]
Thanks, I've updated the module name.

**Comment by vaibhavsagar** [comment]
Thanks!

**Pull request merged**